### PR TITLE
Fix S1075 FN: Support collections and collection expressions

### LINKED ISSUE #8172: Fix S1075 FN: Support collections and collection expressions
The rule should raise an issue when URIs or path delimiters are hardcoded as an element of a collection.
```cs
IList<string> uriList1 = new[] { "C:/test.txt" }; // FN
IList<string> uriList2 = ["C:/test.txt"]; // FN
string[][] uriMatrix1 = [["C:/test.txt"]]; // FN
IDictionary<string, string> uriDict = new Dictionary<string, string> { ["a"] = "C:/test.txt" }; // FN
```

### PR DESCRIPTION
**Pull request by cristian-ambrosini-sonarsource**
Fixes #8172

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs [hunk 1]
```diff
@@ -49,6 +49,7 @@ public abstract class UriShouldNotBeHardcodedBase<TSyntaxKind, TLiteralExpressio
                 "FILE",
                 "PATH",
                 "URI",
+                "URIS",
```
Comment (martin-strecker-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:52]: What about the pluralization of the other `CheckedVariableNames`? Couldn't we do a type lookup and if the type implements `IEnumerable` we allow an `s` at the end (files, paths, uris, and so on) of the variable name. This should also work for the `var uris = new List<string> { "C:/test.txt" }` cases.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]

File: analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs (hunk 1 above)
Comment (cristian-ambrosini-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:52]: Is it really necessary?
I would not consider `var uris = "www.myuri.com";` a FP, so I'm not sure I see the value of additional checks that might or might not slow down the rule. What about `var uri = new List<string> { "C:/test.txt" }`, should we consider it compliant?

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:54]: URLS, URNS, and STREAMS are missing

Comment (cristian-ambrosini-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:54]: Yes, but would you consider `var uris = "www.myuri.com";` a FP?

Comment (martin-strecker-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:54]: Yes. But I also wouldn't care too much about it. You can fix the issue like you did by simply adding "URIS".

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
It seems you missed to accept the this change:
https://github.com/SonarSource/sonar-dotnet/pull/8252#pullrequestreview-1697586390

otherwise looks good.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM.

**Comment by cristian-ambrosini-sonarsource** [comment]
Peach validation: no changes

**Pull request merged**